client_test.go

```diff
@@ -16,6 +16,7 @@ import (
 	apns "github.com/sideshow/apns2"
 	"github.com/sideshow/apns2/certificate"
 	"github.com/stretchr/testify/assert"
+	"golang.org/x/net/context"
```

Comment:
It would be good to see two versions of the context tests in separate files with appropriate build tags, just to ensure that it works with either package.